### ISSUE TITLE
[defect]: SQLite Data Usage Accounting not working 

### ISSUE THREAD

**Issue by hardillb** [FreeRADIUS/freeradius-server#4198]
### What type of defect/bug is this?

Unexpected behaviour (obvious or verified by project member)

### How can the issue be reproduced?

Follow these instructions but use sqlite instead of mysql or postgress
https://wiki.freeradius.org/guide/Data-Usage-Reporting

Then run the script to generate new usage periods

./process-radacct-refresh.sh freeradius.db

### Log output from the FreeRADIUS daemon

```shell
NA this is a problem with the SQL schema/script not the core daemon.
```

### Relevant log output from client utilities

With the current script the output is as follows:
```
sqlite> select * from data_usage_by_period ;
username|period_start|period_end|acctinputoctets|acctoutputoctets
isp1|1970-01-01 00:00:00|2021-08-18 23:10:31|501459|311127
isp1|2021-08-18 23:10:32||-501459|-311127
isp1|2021-08-18 23:10:33|2021-08-18 23:13:21|501459|311127
isp1|2021-08-18 23:13:22||-501459|-311127
isp1|2021-08-18 23:13:23|2021-08-18 23:16:00|541238|446748
isp1|2021-08-18 23:16:01||-541238|-446748
isp1|2021-08-18 23:16:02|2021-08-19 11:42:52|665474|577812
isp1|2021-08-19 11:42:53||-665474|-577812
isp1||2021-08-19 12:48:54|675542|588456
isp1|2021-08-19 12:48:55||-675542|-588456
```

This is because the script is adding 1 second to the `period_start` value so it doesn't match the previous value with no `period_end` value

Fix for this in PR #4196 

Looks like it should be applied to the 4.0 and 3.0 branches

**Comment by jpereira** [comment]
Hi @hardillb, thank you for the patch. It will be checked as soon as we get a chance.

**Comment by alandekok** [comment]
Fixed in #4199.
